programs/mango-v4/src/state/mango_account.rs

```diff
@@ -1857,19 +2078,40 @@ mod tests {
 
     fn make_test_account() -> MangoAccountValue {
         let account = MangoAccount::default_for_tests();
-        let bytes = AnchorSerialize::try_to_vec(&account).unwrap();
+        let mut bytes = AnchorSerialize::try_to_vec(&account).unwrap();
 
+        // The MangoAccount struct is missing some dynamic fields, add space for them
+        let obv2_length = 5;
         // Verify that the size is as expected
         let expected_space = MangoAccount::space(
             account.tokens.len() as u8,
             account.serum3.len() as u8,
             account.perps.len() as u8,
             account.perp_open_orders.len() as u8,
             account.token_conditional_swaps.len() as u8,
+            obv2_length,
         );
+        bytes.extend(vec![0u8; expected_space - bytes.len()]);
         assert_eq!(expected_space, 8 + bytes.len());
+        // Set the length of these dynamic parts
+        let (fixed, dynamic) = bytes.split_at_mut(size_of::<MangoAccountFixed>());
+        let mut header = MangoAccountDynamicHeader::from_bytes(dynamic).unwrap();
+        header.token_conditional_swap_count = account.token_conditional_swaps.len() as u8;
+        header.openbook_v2_count = obv2_length;
+        let mut account = MangoAccountRefMut {
+            header: &mut header,
+            fixed: bytemuck::from_bytes_mut(fixed),
+            dynamic,
+        };
+        account.write_token_conditional_swap_length();
+        account.write_openbook_v2_length();
+        let mut account = MangoAccountValue::from_bytes(&bytes).unwrap();
+        // Initialize the openbook orders with defaults as they would be in the program
+        for i in 0..header.openbook_v2_count() {
+            *account.openbook_v2_orders_mut_by_raw_index(i) = OpenbookV2Orders::default();
+        }
```

Comment:
now that you've changed `MangoAccount` most of the modifications to this function can go away